js/main.js

```diff
@@ -92,3 +123,20 @@ function grabView(event) {
   var displayView = event.target.getAttribute('data-view');
   switchViews(displayView);
 }
+
+$unorderedList.addEventListener('click', editIconHandler);
+function editIconHandler(event) {
+  if (event.target.tagName === 'I') {
+    var grabEditEntryId = parseInt(event.target.closest('li').getAttribute('data-entry-id'));
+    data.editing = grabEditEntryId;
+    for (var i = 0; i < data.entries.length; i++) {
+      if (grabEditEntryId === data.entries[i].entryId) {
+        $title.value = data.entries[i].title;
+        $inputPhotoUrl.value = data.entries[i].photoURL;
+        $notes.value = data.entries[i].notes;
+        $image.setAttribute('src', $inputPhotoUrl.value);
+      }
+      switchViews('entry-form');
```

Comment:
You should do this below the loop, right now it's happening on every iteration of your loop.